### PR TITLE
JBPM-6178 Stunner - Docking issues

### PR DESCRIPTION
**Pull request by tiagodolphine**
- Updating the WiresDockingControl to support docking position update during resizing.
- Added support to dock nodes on magnets, including adjust position when dragging over boundary.

Related JIRA https://issues.jboss.org/browse/JBPM-6178

@romartin 
@LuboTerifaj 
@hasys

### CODE REVIEW

**Comment by tiagodolphine** [comment]
Jenkins retest this

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java [hunk 1]
```diff
@@ -158,20 +174,71 @@ private boolean _isAccept() {
...
+        final MagnetManager.Magnets magnets = parentShape.getMagnets();
+        final double shapeWidth = shape.getPath().getBoundingBox().getMaxX();
+        final double shapeHeight = shape.getPath().getBoundingBox().getMaxY();
+
+        int magnetIndex = -1;
+        Double minDistance = null;
+        //not considering the zero magnet, that is the center.
+        for (int i = 1; i < magnets.size(); i++) {
+            WiresMagnet magnet = magnets.getMagnet(i);
+            final Point2D shapeLocation = shape.getComputedLocation();
```
Comment (romartin) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java]: calls to `getComputedLocation()` are a bit more "expensive", they consider nested containers and are recursively obtaining the `location` attribute for each parent node, so better moving this outside the loop, makes sense?

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java [hunk 2]
```diff
@@ -158,20 +174,71 @@ private boolean _isAccept() {
...
+            final double magnetX = magnet.getControl().getLocation().getX();
+            final double magnetY = magnet.getControl().getLocation().getY();
+            double distance = abs((abs(shapeLocation.getX()) - abs(magnetX)))
```
Comment (romartin) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java]: Not sure if really makes sense using the shape's width/height... I mean I would consider always the shape's bounding box center point as reference, which you can obtain outside the loop by calculating once, and then by just doing trigonometry calculate distance between each of the magnets point... you can use methods already present in the `Geometry` class, for example `Geometry.distance()`, no need to do the operations again here. 
Anyway just brainstorming, not sure if that makes sense or the implications for it, you know better :) Thx

**Comment by tiagodolphine** [comment]
@romartin hey, good points about your comment, I have applied them I think now it has a better performance and it is simpler.  
Thanks !

**Comment by tiagodolphine** [comment]
@LuboTerifaj @romartin just to remember you, this PR was updated due to the modification on Stunner.

**Pull request merged**